### ISSUE TITLE
Add early stopping

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/ignite#133]
I would like to propose an early stopping handler.
Usage:
```
    from ignite.engines import Engine, Events
    from ignite.handlers import EarlyStopping

    def score_function(engine):
        val_loss = engine.state.metrics['nll']
        return -val_loss

    handler = EarlyStopping(patience=10, score_function=score_function, verbose_logger=print)
    evaluator.add_event_handler(Events.COMPLETED, handler, trainer)
```
[Code](https://github.com/pytorch/ignite/compare/master...vfdev-5:feature_early_stopping)

What do you guys think about ?

**Comment by alykhantejani** [comment]
Sounds like a useful feature, please go ahead with a PR! 

My only nit would be around the logger, instead we should do something like what is done in the [engine](https://github.com/pytorch/ignite/blob/master/ignite/engines/engine.py#L41) and use python's `logging`, this way users can attach their log handlers to the root `ignite` logger

**Comment by vfdev-5** [comment]
@alykhantejani Thanks!  I'll fix the logger and create a PR. Do you mean to add : 
```
self._logger = logging.getLogger(__name__ + "." + self.__class__.__name__)
self._logger.addHandler(logging.NullHandler())
```
into the class ? So, no need to add the option `verbose`.

**Comment by alykhantejani** [comment]
yup, exactly :)

**Comment by alykhantejani** [comment]
closed in #136
